ci/check-each-commit.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/sh
+if [ "$1" = "" ]; then
+	echo "USAGE: $0 remote/head_branch"
+	echo "eg $0 upstream/main"
+	exit 1
+fi
+
+set -e
+set -x
+
+if [ "$(git log --pretty="%H %D" | grep "^[0-9a-f]*.* $1")" = "" ]; then
+	echo "It seems like the current checked-out commit is not based on $1"
+	exit 1
+fi
+for COMMITHASH in `git log --format=format:%H $1...HEAD`; do
```

Comment:
I found this more useful with `git log --reverse` so that the tests are done in the same order as the code was written.